hypervisor/virtualbox/virtualbox.go

```diff
@@ -129,36 +131,52 @@ func (*virtualbox) Modify(vmName string, cpus int, mem int) error {
 	return nil
 }
 
-func (*virtualbox) GetVMProperty(vmName, property string) (*string, error) {
-	prop, ok := propertyMap[property]
+func (*virtualbox) HealthCheck(vmName, healthProperty string) (*string, error) {
```

Comment:
I suggest this to be a private function related to VirtualBox pkg, and to be called directly from inside getVmProperty. and so, it will take one parameter only, the vmName, as the heathProperty is always "status" in the case of VirtualBox.